### PR TITLE
SW-5314: Adds client tab functionality

### PR DESCRIPTION
**Pull request by denialanderror**

### CODE REVIEW

File: internal/server/list_clients.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import (
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/ministryofjustice/opg-sirius-supervision-pro-deputy-hub/internal/sirius"
+)
+
+type DeputyHubClientInformation interface {
```
Comment (gurps-singh) [inline internal/server/list_clients.go:12]: Rename to ProHubClientInformation?

File: internal/server/list_clients.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	XSRFToken            string
+	AriaSorting          sirius.AriaSorting
+	DeputyClientsDetails sirius.DeputyClientDetails
+	ProDeputyDetails        sirius.ProDeputyDetails
+	Error                string
+	ErrorMessage         string
+	Errors               sirius.ValidationErrors
+}
+
+func renderTemplateForClientTab(client DeputyHubClientInformation, tmpl Template) Handler {
```
Comment (gurps-singh) [inline internal/server/list_clients.go:28]: Rename here

File: internal/server/list_clients_test.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/ministryofjustice/opg-sirius-supervision-pro-deputy-hub/internal/sirius"
+	"github.com/stretchr/testify/assert"
+)
+
+type mockDeputyHubClientInformation struct {
```
Comment (gurps-singh) [inline internal/server/list_clients_test.go:12]: Rename as this is Pro Hub and not Deputy Hub

File: internal/server/list_clients_test.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+	count            int
+	lastCtx          sirius.Context
+	err              error
+	deputyData       sirius.ProDeputyDetails
+	deputyClientData sirius.DeputyClientDetails
+	ariaSorting      sirius.AriaSorting
+}
+
+func (m *mockDeputyHubClientInformation) GetProDeputyDetails(ctx sirius.Context, deputyId int) (sirius.ProDeputyDetails, error) {
```
Comment (gurps-singh) [inline internal/server/list_clients_test.go:21]: Rename here

File: internal/server/list_clients_test.go [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+	m.count += 1
+	m.lastCtx = ctx
+
+	return m.deputyData, m.err
+}
+
+func (m *mockDeputyHubClientInformation) GetDeputyClients(ctx sirius.Context, deputyId int, columnBeingSorted string, sortOrder string) (sirius.DeputyClientDetails, sirius.AriaSorting, error) {
```
Comment (gurps-singh) [inline internal/server/list_clients_test.go:28]: And here

File: internal/server/list_clients_test.go [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+	m.count += 1
+	m.lastCtx = ctx
+
+	return m.deputyClientData, m.ariaSorting, m.err
+}
+
+func TestNavigateToClientTab(t *testing.T) {
+	assert := assert.New(t)
+
+	client := &mockDeputyHubClientInformation{}
```
Comment (gurps-singh) [inline internal/server/list_clients_test.go:38]: and here

**Review by MishNajam — APPROVED** [review]

**Pull request merged**